packages/subscription/src/stream.ts

```diff
@@ -21,12 +22,7 @@ export interface Stream<T, TReturn = undefined> extends OperationIterable<T, TRe
   collect(): Operation<Iterator<T, TReturn>>;
   toArray(): Operation<T[]>;
   subscribe(scope: Task): Subscription<T, TReturn>;
-  buffer(scope: Task): Stream<T, TReturn>;
-  stringBuffer(scope: Task): StringBufferStream<TReturn>;
-}
-
-export interface StringBufferStream<TReturn = undefined> extends Stream<string, TReturn> {
-  value: string;
+  buffer(limit?: number): Resource<Stream<T, TReturn>>;
```

Comment:
I'm not a fan of making the default a potential footgun. I'd rather make the limit a mandatory parameter, and if you want to opt in to a completely unbounded buffer, then you can still do so, but you have to call it out.

```ts
let lines = yield app.stdout.lines.buffer(Infinity);
```

Reply:
I would personally not come up with the idea of passing in `Infitity` here. We can call out that possibility in the documentation, but it's at least not very intuitive. There are many streams which don't ever emit very many values and for those streams, using an infinite buffer is just fine. I don't see this as that serious of a foot gun, so I'd prefer the more intuitive API.

Reply:
What about making the limit default to `Infinity` instead of being nullable? I think that would be a good compromise that allows you to easily create infinite buffers, but the typedocs, and the auto-complete will keep the potential pitfalls visible in your peripheral vision.